app/controllers/collection_templates_controller.rb

```diff
@@ -0,0 +1,121 @@
+# frozen_string_literal: true
+
+class CollectionTemplatesController < CollectionDraftsController
+  include CollectionsCmrHelper
+  before_action :templates_enabled?
+  before_action :set_resource, only: %i[create_draft destroy edit update show]
+  before_action :load_umm_schema, only: %i[new edit show new_from_existing]
+
+  def create_draft
+    authorize get_resource
+    draft = CollectionDraft.create_from_template(get_resource, current_user)
+    Rails.logger.info("Audit Log: Collection Draft #{draft.entry_title} was created from a template by #{current_user.urs_uid} in provider: #{current_user.provider_id}")
+    redirect_to edit_collection_draft_path(draft)
+  end
+
+  def new
+    super
+
+    @template_names = names_list
+  end
+
+  def create
+    if names_list.include?(params[:draft][:template_name])
+      flash[:error] = 'A template with that name already exists.'
+      redirect_to manage_collections_path
+      return
+    end
+    # Need to fetch and merge the old data in because the params[:draft] only has the form.
+    # If new submitted no origin, then it's a fresh draft, and the provided draft can be
+    # used.  Otherwise, the data provided by the user needs to be merged with the source data.
+    new_draft = if params[:origin] == ''
+                  params[:draft]
+                else
+                  fetch_source_data.deep_merge(params[:draft])
+                end
+    set_resource(resource_class.new(user: current_user, provider_id: current_user.provider_id, draft: new_draft.to_camel_keys))
+
+    super
+  end
+
+  def new_from_existing
+    set_resource(resource_class.new(user: current_user, provider_id: current_user.provider_id, draft: fetch_source_data))
+    authorize get_resource
+
+    @draft_forms = CollectionDraft.forms
```

Comment:
We are making changes like this for MMT-1667. I think that `@draft_forms` (plural and singular) should be changed to the corresponding `@forms`, and `CollectionDraft` should be changed to the generic method to call the class, as the forms should be inherited.